### ISSUE TITLE
Fix direct API server access in AKS

### ISSUE THREAD

**Issue by weibeld** [weibeld/k8s-conncheck#1]
Detecting the API server URL in AKS works as expected, but the kubectl commands then fail with the following error:

```
Unable to connect to the server: x509: certificate is valid for 10.0.0.1, not 40.114.181.77
```
`40.114.181.77` is the IP address of the API server and `10.0.0.1` is the ClusterIP of the `default/kubernetes` Service.

That means, the API server certificate in AKS is configured for being queried through the `kubernetes` Service and not directly through the API server URL.

To directly access the API server despite this, the `--insecure-skip-tls-verify` flag can be set for the kubectl commands.

The `--insecure-skip-tls-verify` flag cannot be used together with the `--certificate-authority` flag (which is currently used for all kubectl commands).

The best solution is probably to fall back to an alternative kubectl wrapper function that uses the  `--insecure-skip-tls-verify` flag but not the `--certificate-authority` flag when the above error is detected.

**Comment by weibeld** [comment]
The API server IP address detected by the init Pod is the IP address that the AKS API server public domain name resolves to.

For example:

```
Detected API server URL: https://40.114.181.77:443
```

```bash
$ kubectl cluster-info
Kubernetes master is running at https://test-default-9c3ec6-e0f34013.hcp.westeurope.azmk8s.io:443
```

```bash
$ host test-default-9c3ec6-e0f34013.hcp.westeurope.azmk8s.io
test-default-9c3ec6-e0f34013.hcp.westeurope.azmk8s.io has address 40.114.181.77
```

So, the API server certificate on AKS is probably issued for the API server domain name (for access from outside the cluster) and the ClusterIP of the `default/kubernetes` Service (for access from inside the cluster), but not for the IP address of the API server.

**Comment by weibeld** [comment]
Fixed in 0b1d902efa476818fb4d9ea697c166467739cd95
